lib/galaxy/jobs/runners/drmaauniva.py

```diff
@@ -0,0 +1,640 @@
+                  Rs, Rts, RS, RtS, RT, RtT 	all suspended states with re-submit
+        Error 	Eqw, Ehqw, EhRqw 	all pending states with error == GE tried to execute a job in a queue, but it failed for a reason that is specific to the job
+        Deleted 	dr, dt, dRr, ds, dS, dT, dRs, dRS, dRT 	all running and suspended states with deletion, i.e. these jobs have been deleted, but have not yet been purged from the system
+
+        the qstat -j ID states are
+        d(eletion),             qdel has been used
+        E(rror),                pending jobs hat couldn't be started due to job properties
+        h(old),                 job currently is not eligible for execution due to a hold state assigned to it
+        r(unning),              job is about to be executed or is already executing
+        R(estarted),            the job was restarted. This can be caused by a job migration or because of one of the reasons described in the -r section
+        s(uspended),            caused by suspending the job via qmod -s
+        S(uspended),            either that the queue containing the job is suspended ... or that a pending job got suspended due to a preemptive action that was either triggered automatically by the system or by a manual preemption request triggered via qmod(1) -p ... S
+        e(N)hanced suspended,   preemptive states shown for pending jobs triggered either automatically or via qmod(1) -p ... N or -p ... P
+        (P)reempted,            -"-
+        t(ransfering),          job is about to be executed or is already executing
+        T(hreshold) or          shows that at least one suspend threshold of the corresponding queue was exceeded (see queue_conf(5)) and that the job has been suspended as a consequence
+        w(aiting).              or that the job is waiting for completion of the jobs to which job dependencies have been assigned to the job
+
+        job_id: the jobid (for logging)
+        state: a state from the qstat output
+        return: one of the drmaa states
+        """
+        if "d" in state:
+            extinfo["deleted"] = True
+            return self.drmaa.JobState.FAILED
+        if "E" in state:
+            return self.drmaa.JobState.FAILED
+        elif "s" in state:
+            return self.drmaa.JobState.USER_SUSPENDED
+        elif "S" in state or "T" in state or 'N' in state or 'P' in state:
+            return self.drmaa.JobState.SYSTEM_SUSPENDED
+        elif "h" in state:
+            return self.drmaa.JobState.USER_SYSTEM_ON_HOLD
+        elif "r" in state.lower() or "t" in state:
+            return self.drmaa.JobState.RUNNING
+        elif "w" in state:
+            return self.drmaa.JobState.QUEUED_ACTIVE
+        else:
+            logging.error("DRMAAUniva: job {job_id} unknown state from qstat: {state}".format(job_id=job_id, state=state))
+            return self.drmaa.JobState.UNDETERMINED
+
+
+def _check_memory_limit( efile_path ):
```

Comment:
Rather than copying this from `SlurmJobRunner`, it'd probably be better to move it to somewhere like `galaxy.util`, remove the references to memory, and call it `grep_tail` or similar, and just pass in the set of strings to look for.

Reply:
good idea: created `grep_tail` in `galaxy.util` and adapted the `SLURMRunner` accordingly.